src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -266,17 +266,11 @@ public boolean getLocationPermissions() {
 		boolean hasCoarseLocation = ContextCompat.checkSelfPermission(this, ACCESS_COARSE_LOCATION) == PERMISSION_GRANTED;
 
 		if (hasFineLocation && hasCoarseLocation) {
-			trace(this, "getLocationPermissions() :: already granted");
+			trace(this, "getLocationPermissions() :: Fine and Coarse location already granted");
 			return true;
 		}
 
-		if (settings.hasUserDeniedGeolocation()) {
```

Comment:
Does this mean we are never going to stop asking the user for location permissions (even if they have explicitly denied them)?

Reply:
Oh wait, IIRC Android will stop asking after the user has denied (2 times?) right?  But would we still be popping the prominent disclosure every time?

Reply:
It will behave the same as Storage and Send SMS permissions, if the user hasn't granted the permissions, then we try again asking them in case the have change their mind. Ideally we would like to have the GPS anyways for the forms reports. 
If they have denied too many times or have selected "Never ask again", we still ask them to grant the permission and redirect them to the app's settings page so they grant it manually because at the point we can't do it automatically anymore. 

The difference here is that now we have approximate and accurate location, I'm going to with asking for accurate location even if they just granted approximate, it won't block them to use the forms, they can keep choosing "approximate" 🤔